RecoJets/JetProducers/plugins/MVAJetPuIdProducer.cc

```diff
@@ -0,0 +1,339 @@
+#include <memory>
+#include "FWCore/Framework/interface/Frameworkfwd.h" 
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "RecoJets/JetProducers/interface/MVAJetPuId.h"
+
+#include "DataFormats/JetReco/interface/Jet.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/JetReco/interface/PileupJetIdentifier.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "CondFormats/JetMETObjects/interface/JetCorrectorParameters.h"
+#include "CondFormats/JetMETObjects/interface/FactorizedJetCorrector.h"
+#include "JetMETCorrections/Objects/interface/JetCorrectionsRecord.h"
+
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+
+
+class MVAJetPuIdProducer : public edm::stream::EDProducer <> {
```

Comment:
@slava77 TMVAReader is thread-friendly (i.e. different instances of it do not interfere with one another) but it is not thread-safe (i.e. different threads can not safely call the same instance of the class). stream modules only need the thread-friendly guarantee. global modules need the thread-safe guarantee (technically ```const``` thread-safe guarantee).